File: Planner-Gui/app/src/main/cpp/event.cpp
#include "include/event.hpp"

Event::Event(const char *filepath, std::string date, std::string title, std::string description,
             int startTime, int duration) {
    this->description = description;
    this->title = title;
    this->startTime = startTime;
    this->duration = duration;
    std::string time = numToString(startTime);
    std::string durations = numToString(duration); 
   
 
    const char* day = date.substr(0,2).c_str();
    const char* month = date.substr(3,2).c_str();
    const char *year =date.substr(6,4).c_str();
    if(checkDate(day, month, year)){
        bool q = addEvent( filepath,day, month, year, title.c_str(), description.c_str(), time.c_str(), durations.c_str());
    }
    else{
        bool x = createDate(day, month, year);
        bool y = addEvent(filepath, day, month, year, title.c_str(), description.c_str(), time.c_str(), durations.c_str());
    }
        
}
Event::Event(std::string title, std::string description, int startTime, int duration){

    this->title = title;
    this->description = description;
    this->startTime = startTime;
    this->duration = duration;
}
std::string Event::toString() {
	std::string result = "";
	result = ";" + title + ";" + description + ";" +
			numToString(startTime) + ";"+ numToString(duration) +":";
	return result;
}

File: Planner-Gui/app/src/main/java/com/kwrp/planner_gui/Event.java
package com.kwrp.planner_gui;

import android.util.Log;


/**
 * A class defining an event object containing all the appropriate
 * details (e.g. title, description)
 *
 * @author KWRP
 */
public class Event {
    /*
     * The event ID
     */
    private String eventId;
    private String eventDate;

    /*
     * The event title
     */
    private String title;

    /*
     * The event description
     */
    private String description;

    /*
     * The event start time
     */
    private String startTime;

    /*
     * The event duration
     */
    private String finishTime;
    private String endDate;


    /**
     * Constructor given an event formatted in a standard string form
     *
     * @param event the string event
     */
    public Event(String event) {

        Log.e("s", event);
        String[] eventItems = event.split("__");
        String[] startDay = eventItems[0].split("-");
        this.eventDate = startDay[2] + "/" + startDay[1] + "/" + startDay[0];
        this.title = eventItems[1];
        this.description = eventItems[2];
        this.startTime = eventItems[3];
        this.finishTime = eventItems[4];
        String[] day = eventItems[5].split("-");
        this.endDate = day[2] + "/" + day[1] + "/" + day[0];
        this.eventId = eventItems[6];
    }

    /**
     * Gets the eventId
     *
     * @return the eventId
     */
    public String getEventId() {
        return eventId;
    }

    /** Returns the end date
     *
     * @return String the end date
     */
    public String getEndDate() { return endDate;}

    /**
     * Get the event title
     *
     * @return the event title
     */
    public String getTitle() {
        return title;
    }

    /**
     * Get the event description
     *
     * @return the event description
     */
    public String getDescription() {
        return description;
    }

    /** Gets the finish time
     *
     * @return String finish time
     */
    public String getFinishTime() { return finishTime;}

    /**
     * Get the event start time
     *
     * @return the event start time
     */
    public String getStartTime() {
        return startTime;
    }

    /**
     * Converts object to string
     *
     * @return the string version of the object displaying all fields
     */
    @Override
    public String toString() {
        return "\t\t" + title + " starts at " + startTime +
                "\n\t\t"  +description;

    }
}
